apps/site/test/site_web/views/trip_plan_view_test.exs

```diff
@@ -1435,4 +1435,442 @@ closest arrival to 12:00 AM, Thursday, January 1st."
+            additional_valid_modes: [:bus],
+            cents: 9000,
+            duration: :month,
+            media: [:charlie_card, :charlie_ticket],
+            mode: :subway,
+            name: :subway,
+            price_label: nil,
+            reduced: nil
+          },
+          recommended_month_pass: %Fares.Fare{
+            additional_valid_modes: [:bus],
+            cents: 9000,
+            duration: :month,
+            media: [:charlie_card, :charlie_ticket],
+            mode: :subway,
+            name: :subway,
+            price_label: nil,
+            reduced: nil
+          },
+          reduced_month_pass: %Fares.Fare{
+            additional_valid_modes: [:bus],
+            cents: 3000,
+            duration: :month,
+            media: [:senior_card, :student_card],
+            mode: :subway,
+            name: :subway,
+            price_label: nil,
+            reduced: :any
+          }
+        },
+        start: DateTime.from_unix!(0),
+        stop: DateTime.from_unix!(0)
+      }
+
+      assert show_monthly_passes?(login_sl_plus_subway_itinerary)
+    end
+
+    test "returns true for an itinerary without any transit legs" do
+      no_transit_legs_itinerary = %TripPlan.Itinerary{
+        accessible?: false,
+        legs: [
+          %TripPlan.Leg{
+            description: "WALK",
+            from: %TripPlan.NamedPosition{
+              latitude: 42.365396,
+              longitude: -71.017547,
+              name: "Boston Logan Airport",
+              stop_id: nil
+            },
+            long_name: nil,
+            mode: %TripPlan.PersonalDetail{
+              distance: 385.75800000000004,
+              steps: [
+                %TripPlan.PersonalDetail.Step{
+                  absolute_direction: :southwest,
+                  distance: 382.50800000000004,
+                  relative_direction: :depart,
+                  street_name: "footbridge"
+                },
+                %TripPlan.PersonalDetail.Step{
+                  absolute_direction: :northwest,
+                  distance: 3.25,
+                  relative_direction: :left,
+                  street_name: "service road"
+                }
+              ]
+            },
+            name: "",
+            to: %TripPlan.NamedPosition{
+              latitude: 42.364612,
+              longitude: -71.020862,
+              name: "Terminal A",
+              stop_id: "17091"
+            },
+            type: nil,
+            url: nil
+          },
+          %TripPlan.Leg{
+            description: "WALK",
+            from: %TripPlan.NamedPosition{
+              latitude: 42.352271,
+              longitude: -71.055242,
+              name: "South Station",
+              stop_id: "74617"
+            },
+            long_name: nil,
+            mode: %TripPlan.PersonalDetail{
+              distance: 0.0,
+              steps: [
+                %TripPlan.PersonalDetail.Step{
```

Comment:
It's not a problem, but generally speaking I don't think you need quite this much detail in your test data? For example `%TripPlan.PersonalDetail.Step{}` will just use the defstruct values. [An example of taking advantage of this for a test](https://github.com/mbta/dotcom/blob/eaac9fdb9b32f0c55f64689ec23de01de283a513/apps/trip_plan/test/itinerary_test.exs#L88).